css/src/components/form/checkbox.scss

```diff
@@ -87,9 +90,30 @@ $checkbox-spacing: $spacer-3 !default;
 			}
 		}
 
-		&[disabled] + .checkbox-check {
-			opacity: 0.6;
-			cursor: not-allowed;
+		&[disabled] {
+			+ .checkbox-check {
+				border-color: $checkbox-unchecked-border-color;
+				opacity: 0.6;
+			}
+
+			+ .checkbox-check,
+			~ .checkbox-text {
+				cursor: not-allowed;
+			}
+		}
+
+		&:focus-visible {
+			+ .checkbox-check {
+				@extend %focus;
+			}
+
+			&:not(:checked) + .checkbox-check {
```

Comment:
Do we need this rule for special box shadow focus? Seems like global focus styles would be more robust.

Reply:
Updated focus styles